sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4574,6 +4574,22 @@ def _symlink_node(runner: command_runner.CommandRunner):
                     f'may already exist. Log: {log_path}')
 
             subprocess_utils.run_in_parallel(_symlink_node, runners)
+        #in case of compression, inflate all zip files
+        # can optimize further by going to specific filemounts & workdir only
+        def _decompress_filemount_zips(runner: command_runner.CommandRunner):
+            zip_filename = (f'{constants.SKY_REMOTE_WORKDIR}'
+                            '/skypilot-filemounts.zip')
+            decompress_command = (
+                f'[ -f {zip_filename} ] && '
+                f'(unzip {zip_filename} -d {constants.SKY_REMOTE_WORKDIR} '
+                f'&& rm {zip_filename}) || echo "Zip not on this node"')
+            returncode = runner.run(decompress_command, log_path=log_path)
+            subprocess_utils.handle_returncode(
+                returncode, decompress_command,
+                'Failed to inflate or remove skypilot-filemounts.zip, '
+                f'check permissions. Log: {log_path}')
```

Comment:
We might want to have a list of reserved names for those compressed file, as it is possible that the user's workdir have a file named `skypilot-filemounts.zip` and causing a naming conflict. We should check first and raise an error if there are such a file.

Reply:
Also, we could attach some uuid hash for this run to make it harder to conflict.